src/component/Calculator.js

```diff
@@ -67,44 +67,40 @@ const buttons = [
   },
 ];
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
+const Calculator = () => {
+  const [state, setState] = useState({});
 
-    this.state = {};
-  }
+  const onClickHandler = useCallback((e) => {
+    setState((nextState) => {
+      const display = calculate(nextState, e.target.value);
+      return ({
+        ...nextState,
+        ...display,
+      });
+    });
+  }, []);
 
-  onClickHandler = (e) => {
-    this.setState((nextState) => calculate(nextState, e.target.value));
-  };
+  const { total, next, operation } = state;
+  const displayNum = (total || '') + (operation || '') + (next || '');
 
-  displayOutput = () => {
-    const { total, next, operation } = this.state;
-    const displayNum = (total || '') + (operation || '') + (next || '');
-
-    return displayNum || '0';
-  };
-
-  render() {
-    return (
-      <div className="calculator-board">
-        <div className="output-box">
-          <p>{this.displayOutput()}</p>
-        </div>
-        <div className="grid">
-          {buttons.map(({ text, className }) => (
-            <input
-              value={text}
-              className={`button grid-item ${className || ''}`}
-              key={text}
-              type="button"
-              onClick={this.onClickHandler}
-            />
-          ))}
-        </div>
+  return (
+    <div className="calculator-board">
+      <div className="output-box">
+        <p>{displayNum || '0'}</p>
+      </div>
```

Comment:
- [ ] There are errors on the browser when a number is divided by zero or when an operator is clicked twice. As your app should work without errors please incorporate some error handling for `displayNum`. 
![Screenshot from 2021-08-12 22-50-28](https://user-images.githubusercontent.com/64894927/129236603-6b50938b-9455-4585-b5a1-9d6531467dc1.png)
![Screenshot from 2021-08-12 22-50-18](https://user-images.githubusercontent.com/64894927/129236604-5f07f120-0859-4f70-88e1-9605ccd9a36a.png)